Assets/SteamVR_Unity_Toolkit/Scripts/VRTK_PlayerClimb.cs

```diff
@@ -0,0 +1,201 @@
+﻿namespace VRTK
+{
+    using UnityEngine;
+    using System.Collections;
+
+    public struct PlayerClimbEventArgs
+    {
+        public uint controllerIndex;
+        public GameObject target;
+    }
+
+    public delegate void PlayerClimbEventHandler(object sender, PlayerClimbEventArgs e);
+
+    public class VRTK_PlayerClimb : MonoBehaviour
+    {
+        public event PlayerClimbEventHandler PlayerClimbStarted;
+        public event PlayerClimbEventHandler PlayerClimbEnded;
+
+        public bool usePlayerScale = true;
+        public bool useGravity = true;
+
+        private Transform controllerTransform;
+        private Vector3 startControllerPosition;
+        private Vector3 startPosition;
+
+        private Vector3 lastGoodHeadsetPosition;
+        private bool headsetColliding = false;
+        private bool isClimbing = false;
+
+        private VRTK_PlayerPresence playerPresence;
+        private bool lastGravitySetting;
+
+        public virtual void OnPlayerClimbStarted(PlayerClimbEventArgs e)
```

Comment:
nevermind, this is a unity event isn't it :)

You'll need to document any events thrown in the "Class Events" section of the doc (look at another script that has a Class Events section)

You list what it does and the event payload it's sending